docs/aws/audit/iamcompliance/rules/user_account_inline_policies.mdx

```diff
@@ -23,6 +23,274 @@ HIPAA, GDPR, CISAWS, CBP, NIST, SOC2, PCIDSS, HITRUST, NISTCSF
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start using AWS CLI, you need to install it on your local machine and configure it with your AWS account credentials. You can do this by running the following commands:
+
+   Installation:
+   ```
+   pip install awscli
+   ```
+   Configuration:
+   ```
+   aws configure
+   ```
+   You will be prompted to provide your AWS Access Key ID, Secret Access Key, default region name, and default output format.
+
+2. List all IAM users: Use the following AWS CLI command to list all IAM users in your AWS account:
+
+   ```
+   aws iam list-users
+   ```
+   This command will return a JSON object with details of all IAM users.
+
+3. For each user, list all inline policies: For each IAM user, you can list all inline policies attached to that user by using the following AWS CLI command:
+
+   ```
+   aws iam list-user-policies --user-name <username>
+   ```
+   Replace `<username>` with the name of the IAM user. This command will return a JSON object with details of all inline policies attached to the specified IAM user.
+
+4. Check for inline policies: If the `list-user-policies` command returns any policies, it means that the user has inline policies attached. This is a misconfiguration as users should not have inline policies. Instead, they should have managed policies which are easier to manage and provide better control over permissions.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3):
+   You need to install Boto3 in your Python environment. You can do this using pip:
+   ```
+   pip install boto3
+   ```
+   After installing Boto3, you need to configure it with your AWS credentials. You can do this by setting the following environment variables:
+   ```
+   AWS_ACCESS_KEY_ID = 'your_access_key'
+   AWS_SECRET_ACCESS_KEY = 'your_secret_key'
+   ```
+
+2. Import necessary libraries and create an IAM client:
+   You need to import Boto3 and create an IAM client to interact with AWS IAM. Here is how you can do it:
+   ```python
+   import boto3
+
+   # Create IAM client
+   iam = boto3.client('iam')
+   ```
+
+3. List all IAM users and their inline policies:
+   You can use the `list_users` method to get all IAM users and the `list_user_policies` method to get all inline policies for each user. Here is how you can do it:
+   ```python
+   # List all users
+   users = iam.list_users()
+   for user in users['Users']:
+       # List all inline policies for each user
+       inline_policies = iam.list_user_policies(UserName=user['UserName'])
+       if inline_policies['PolicyNames']:
+           print(f"User {user['UserName']} has inline policies: {inline_policies['PolicyNames']}")
+   ```
```

Comment:
<h2>Secret Detected: Secret Keyword</h2> <a href="https://console.cloudanix.com/code?t=secrets"> Click here for more details</a>